sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/PartitionMetadata.scala

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.implementation.{
+  CosmosClientMetadataCachesSnapshot,
+  SparkBridgeImplementationInternal
+}
+import com.azure.cosmos.spark.CosmosPredicates.{
+  requireNotNull,
+  requireNotNullOrEmpty
+}
+import org.apache.spark.broadcast.Broadcast
+
+import java.time.Instant
+import java.util.concurrent.atomic.AtomicLong
+
+private object PartitionMetadata {
+  def createKey(databaseId: String,
+                containerId: String,
+                feedRange: String): String =
+    s"$databaseId|$containerId|$feedRange"
```

Comment:
what if the databaseId or containerId has "|" character?

see https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionTest.java#L35

Reply:
Correct - but to make it safer I switched to using "/" as separator - that character is not allowed in container or database names.

Reply:
Fixed in next iteration